### DISCUSSION TITLE
Mark kwargs sometimes not passed into request.node

### DISCUSSION THREAD

**Discussion by AAAver** [pytest-dev/pytest#8530] [Q&A] [answered]
I have some tests where I pass data to fixtures with marker kwargs.
In most of the cases everything goes well, but sometimes kwargs are not passed to fixture. 
On the screenshot below you can see, that I pass assets_num=1 as kwarg to alert_params marker. But when i try to access this kwarg in fixture, there is no kwargs at all.
Normally this test goes well in 9/10 situations, but sometimes not

![Снимок экрана 2021-04-07 в 16 12 08](https://user-images.githubusercontent.com/53422558/113872476-98e93c00-97bc-11eb-97b3-fc45aaabfb2a.png)

**Comment by RonnyPfannschmidt** [comment]
what version of python/pytest ?

**Comment by RonnyPfannschmidt** [comment]
also please print out the value returned from mark.call

on the python cli i get 
```
In [1]: import pytest

In [2]: pytest.mark.alert_params(asset_num=1)
Out[2]: MarkDecorator(mark=Mark(name='alert_params', args=(), kwargs={'asset_num': 1}))
```

so at first glance your issue doesnt make sense

**Comment by AAAver** [comment, marked as answer]
Finally I've found my mistake in code
```
@pytest.fixture()
def some_fixture(request):
    marker = request.node.get_closest_marker("some_fixture_params")
    some_params = {}
    num = None
    if marker:
        if 'num' in marker.kwargs.keys():
            num = marker.kwargs['num']
            marker.kwargs.pop('num')
            some_params.update(marker.kwargs)


@mark.some_fixture_params(num=1)
def test_xxx(some_fixture):
    assert False
```

The case was that for some reasons I had to pop number parameter from kwargs and update other parameter with that that was left in kwargs
So if test failed for some reason and reruns I had already poped number parameter from mark. I did not know that marker uses same instance for rerun.

So there is no error.
